### PR TITLE
Split out some utilities

### PR DESCRIPTION
**Pull request by axic**
And add tests for them.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -25,6 +25,7 @@ static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
 mod native;
 pub mod types;
+mod utils;
```
Comment (s1na) [inline src/lib.rs]: Would you rather keep the existing `unsafe_alloc_buffer` defined in `src/lib.rs`? seems redundant

Comment (axic) [reply src/lib.rs]: Forgot to delete. Reason for this change is #55, which also needs it.

Comment (axic) [reply src/lib.rs]: Fixed.

File: src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#[cfg(feature = "std")]
+pub fn unsafe_alloc_buffer(len: usize) -> Vec<u8> {
```
Comment (axic) [inline src/utils.rs]: Since above we don't use `pub mod utils;` I guess this can't be imported from outside the create.

**Comment by axic** [comment]
We can as we go, though this is the only alloc function right now.

**Pull request merged**